include/lsst/afw/image/PhotoCalib.h

```diff
@@ -0,0 +1,426 @@
+     * @param[in]  countsMag0      The mean of the zeroPoint() over its bounding box.
+     * @param[in]  zeroPoint       The spatially varying photometric zero point.
+     * @param[in]  countsMag0Sigma The error on the zero point.
+     * @param[in]  isConstant      Is this PhotoCalib spatially constant?
+     */
+    PhotoCalib(double countsMag0, double countsMag0Sigma,
+               std::shared_ptr<afw::math::BoundedField> zeroPoint,
+               bool isConstant):
+        _zeroPoint(zeroPoint), _countsMag0(countsMag0), _countsMag0Sigma(countsMag0Sigma),
+        _isConstant(isConstant) {}
+
+    //@{
+    /**
+     * Convert counts in ADU to maggies.
+     *
+     * If passed point, use the exact calculation at that point, otherwise, use the mean scaling factor.
+     *
+     * @param[in]  counts  The source counts in ADU.
+     * @param[in]  point   The point that counts is measured at (must be within the domain of the
+     *                     BoundedField of this PhotoCalib).
+     * @param[in]  throwOnNegativeCounts  Raise an exception when passed negative counts.
+     *
+     * @return     The flux in maggies.
+     */
+    double countsToMaggies(double counts,
+                           const afw::geom::Point<double, 2> &point,
+                           bool throwOnNegativeCounts=false) const;
+    double countsToMaggies(double counts,
+                           bool throwOnNegativeCounts=false) const;
+    //@}
+
+    //@{}
+    /**
+     * Convert counts and error in counts (ADU) to maggies and maggies error.
+     *
+     * If passed point, use the exact calculation at that point, otherwise, use the mean scaling factor.
+     *
+     * @param[in]  counts       The source fluxcounts in ADU.
+     * @param[in]  countsSigma  The counts error (sigma).
+     * @param[in]  point        The point that counts is measured at (must be within the domain of the
+     *                          BoundedField of this PhotoCalib).
+     * @param[in]  throwOnNegativeCounts  Raise an exception when passed negative counts.
+     *
+     * @return     The flux in maggies and error (sigma).
+     */
+    Quantity countsToMaggies(double counts, double countsSigma,
+                             const afw::geom::Point<double, 2> &point,
+                             bool throwOnNegativeCounts=false) const;
+    Quantity countsToMaggies(double counts, double countsSigma,
+                             bool throwOnNegativeCounts=false) const;
+    //@}
+
+    /**
+     * Convert sourceRecord[countsField_counts] (ADU) at location
+     *             (sourceRecord.get('x'), sourceRecord.get('y')) (pixels) to maggies and maggie error.
+     *
+     * @param[in]  sourceRecord  The source record to get counts and position from.
+     * @param[in]  countsField   The counts field: Keys of the form "*_counts" and "*_countsSigma" must exist.
+     *                           For example: countsField = "PsfFlux" -> "PsfFlux_counts", "PsfFlux_countsSigma"
```

Comment:
This definitely depends on the results of RFC-322. If that's rejected, there's no obvious place to put such values. Also, as noted in DM-10155, we might actually want to merge the maggies and magnitudes versions into a single method that sets both values in the catalog.